From 61fc0868699805733a8f4357783bcdd06629d708 Mon Sep 17 00:00:00 2001
From: Nick Vatamaniuc <vatamane@apache.org>
Date: Tue, 9 Jul 2019 16:45:52 -0400
Subject: [PATCH] Add erlang 22 support

Bumped elixir version to 1.7.4 as 1.6.6 wasn't built with Erlang 22
support.

Also moving straight to 22.0.5 since 22.0 in travis crashed with a
segmentation fault. Some of the release comments in the point release
mention VM crashes, so it seems to check out.

Fixes https://github.com/apache/couchdb/issues/2069
---
 .travis.yml         | 3 ++-
 rebar.config.script | 6 ++++--
 2 files changed, 6 insertions(+), 3 deletions(-)
diff --git a/rebar.config.script b/rebar.config.script
index 33ceb72838..6445057e7e 100644
--- a/rebar.config.script
+++ b/rebar.config.script
@@ -44,6 +44,8 @@ case VerList of
     [20 | _] = V20 when V20 >= [20, 3, 8, 11] -> ok;
     [21, 2] -> NotSupported(VerString);
     [21, 2, N | _] when N < 3 -> NotSupported(VerString);
+    [22, 0] -> NotSupported(VerString);
+    [22, 0, N | _] when N < 5 -> NotSupported(VerString);
     _ -> ok
 end.
 
@@ -148,7 +150,7 @@ ErlOpts = case os:getenv("ERL_OPTS") of
 end,
 
 AddConfig = [
-    {require_otp_vsn, "17|18|19|20|21"},
+    {require_otp_vsn, "17|18|19|20|21|22"},
     {deps_dir, "src"},
     {deps, lists:map(MakeDep, DepDescs)},
     {sub_dirs, SubDirs},
diff --git a/rebar.config b/rebar.config
index f134a63..1785295 100644
--- a/src/snappy/rebar.config
+++ b/src/snappy/rebar.config
@@ -1,4 +1,4 @@
-{require_otp_vsn, "R14|R15|R16|17|18|19|20|21"}.
+{require_otp_vsn, "R14|R15|R16|17|18|19|20|21|22"}.
 
 {erl_opts, [debug_info, warn_unused_vars, warn_shadow_vars, warn_unused_import]}.
 {port_sources, ["c_src/*.cc", 
